### ISSUE TITLE
Typescript 

### ISSUE THREAD

**Issue by bdombro** [egeozcan/ppipe#8]
Would you like that?

Very cool library btw!

**Comment by egeozcan** [comment]
Hi, I'm always happy to receive help with the typings, but I don't personally have time to figure it out all on my own 😅

**Comment by bdombro** [comment]
Yeah I tried last night and it’s very hard actually :-/

**Comment by egeozcan** [comment]
yup, same! I'll keep this open with hopes to someone with great TS skills helping^^

**Comment by egeozcan** [comment]
finally achieved this after more than 5 years with the help of claude :)
